litestar/plugins/opentelemetry/instrumentation.py

```diff
@@ -0,0 +1,253 @@
+"""OpenTelemetry instrumentation helpers for Litestar constructs.
+
+This module provides optional instrumentation for Litestar-specific features like guards,
+lifecycle events, dependency injection, and channels. All functions gracefully handle
+cases where OpenTelemetry is not installed.
+"""
+
+from __future__ import annotations
+
+import inspect
+from contextlib import asynccontextmanager, contextmanager
+from functools import wraps
+from typing import TYPE_CHECKING, Any, Callable
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncGenerator, Generator
+
+    from litestar.connection import ASGIConnection
+    from litestar.handlers.base import BaseRouteHandler
+__all__ = (
+    "create_span",
+    "instrument_channel_operation",
+    "instrument_dependency",
+    "instrument_guard",
+    "instrument_lifecycle_event",
+)
+
+# Global flag to track if OTEL is available
+_OTEL_AVAILABLE: bool | None = None
+try:  # pragma: no cover - optional dependency
+    from opentelemetry.trace import TracerProvider as _RuntimeTracerProvider
+except ImportError:  # pragma: no cover
+    _RuntimeTracerProvider = object  # type: ignore[misc,assignment]
+
+_CUSTOM_TRACER_PROVIDER: _RuntimeTracerProvider | None = None  # pyright: ignore
+
+
+def _get_tracer(name: str) -> Any:
+    from opentelemetry import trace
+
+    tracer_provider = _CUSTOM_TRACER_PROVIDER or trace.get_tracer_provider()
+    return tracer_provider.get_tracer(name)
+
+
+def _is_otel_available() -> bool:
+    """Check if OpenTelemetry is installed and available."""
+    global _OTEL_AVAILABLE
+    if _OTEL_AVAILABLE is not None:
+        return _OTEL_AVAILABLE
+
+    try:
+        import opentelemetry  # noqa: F401
+
+        _OTEL_AVAILABLE = True
+    except ImportError:
+        _OTEL_AVAILABLE = False
+
+    return _OTEL_AVAILABLE
+
+
+@contextmanager
+def create_span(name: str, attributes: dict[str, Any] | None = None) -> Generator[Any, None, None]:
+    """Create an OTEL span if OpenTelemetry is available.
+
+    Args:
+        name: The span name
+        attributes: Optional attributes to add to the span
+
+    Yields:
+        The span object if OTEL is available, otherwise None
+    """
+    if not _is_otel_available():
+        yield None
+        return
+
+    from opentelemetry.trace import Status, StatusCode
+
+    tracer = _get_tracer(__name__)
+    with tracer.start_as_current_span(name) as span:
+        if attributes:
+            span.set_attributes(attributes)
+
+        try:
+            yield span
+        except Exception as exc:
```

Comment:
Is this necessary? I was under the impression that the active span already records exceptions